qtdialogs/DlgSetupManager.py

```diff
@@ -0,0 +1,1197 @@
+      walletGrid.setRowStretch(1, 1)
+
+      # Add grid to wallet frame
+      walletFrameLayout.addLayout(walletGrid)
+
+      # Refresh button with center alignment
+      self.refreshButton = QtWidgets.QPushButton(self.tr('Refresh Wallet List'))
+      self.refreshButton.setFixedWidth(200)
+      self.refreshButton.clicked.connect(self.loadWalletList)
+      self.refreshButton.setStyleSheet(UI_STYLE_BUTTON_STANDARD)
+      self.refreshButton.setToolTip(
+         'Click to refresh the list of available wallets.\n\n'
+         'Use this when you:\n'
+         '• Add new wallets to the directory\n'
+         '• Remove wallets from the directory\n'
+         '• Want to ensure the list is up to date'
+      )
+
+      # Add button to wallet frame with center alignment
+      buttonLayout = QtWidgets.QHBoxLayout()
+      buttonLayout.addStretch()
+      buttonLayout.addWidget(self.refreshButton)
+      buttonLayout.addStretch()
+      walletFrameLayout.addLayout(buttonLayout)
+
+      # Add wallet frame to main layout
+      mainLayout.addWidget(walletFrame)
+      mainLayout.addStretch()
+
+      tab.setLayout(mainLayout)
+      return tab
+
+   def loadWalletList(self):
+      """Query bridge for wallets and render actions per type/state."""
+      self.walletList.clear()
+      try:
+         # Ask bridge; if bridge not ready, show a placeholder and exit
+         if (TheBDM is None or
+            not getattr(TheBridge.bridgeSocket, 'run', False) or
+            not TheBridge.bridgeSocket.bip15xConnection.ready()):
+            placeholder = QtWidgets.QTreeWidgetItem()
+            placeholder.setText(0, self.tr('Bridge is not running yet'))
+            placeholder.setText(1, '')
+            placeholder.setText(2, '')
+            placeholder.setText(3, '')
+            self.walletList.addTopLevelItem(placeholder)
+            return
+
+         # List wallets using wallet manager (returns a List of WalletFileData)
+         wltList = TheBridge.wltManager.listWallets()
+         # Iterate by index to satisfy capnp dynamic list reader
+         for idx in range(len(wltList)):
+            entry = wltList[idx]
+            # Derive display values
+            walletPath = getattr(entry, 'path', '')
+            fileName = os.path.basename(walletPath) if walletPath else ''
+            walletId = getattr(entry, 'walletId', '')
+            loadState = getattr(entry, 'state', 0)
+
+            # Create row
+            item = QtWidgets.QTreeWidgetItem()
+            item.setText(0, walletId if walletId else fileName)
+            item.setText(1, getattr(entry, 'title', walletId or fileName))
+
+            # Determine state and action
+            # WalletManagerReply.WalletLoadState: 1=legacy, 3=encrypted, 4=ready
+            # loadState is a capnp enum; rely on underlying numeric value via string compare
+            loadStateStr = str(loadState)
+            isLegacy = (loadStateStr.endswith('.legacy')) or (not bool(walletId))
```

Comment:
there is a straight up field for legacy status, use that instead of string manipulation